packages/rs-dapi-client/src/mock.rs

```diff
@@ -0,0 +1,321 @@
+//! Mock implementation of rs-dapi-client for testing
+//!
+//! rs-dapi-client provides `mocks` feature that makes it possible to mock the transport layer.
+//! Core concept of the mocks is a [MockDapiClient] that mimics [DapiClient] behavior and allows
+//! to define expectations for requests and responses using [`MockDapiClient::expect`] function.
+//!
+//! In order to use the mocking feature, you need to:
+//!
+//! 1. Define your requests and responses.
+//! 2. Create a [MockDapiClient] and use it instead of [DapiClient] in your tests.
+//!
+//! See `tests/mock_dapi_client.rs` for an example.
+
+use std::collections::HashMap;
+use tonic::async_trait;
+
+use crate::{
+    transport::{TransportClient, TransportRequest},
+    Dapi, DapiClientError, RequestSettings,
+};
+
+/// Mock DAPI client.
+///
+/// This is a mock implmeneation of [Dapi] that can be used for testing.
+///
+/// See `tests/mock_dapi_client.rs` for an example.
+#[derive(Default)]
+pub struct MockDapiClient {
+    expectations: HashMap<Vec<u8>, Vec<u8>>,
+}
+impl MockDapiClient {
+    /// Create a new mock client
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Add a new expectation for a request
+    pub fn expect<R>(&mut self, request: &R, response: &R::Response)
```

Comment:
If a repetitive configuration may happen it's better to return `&mut Self` (or take `mut self` and return `Self`), then it's possible to do many `expect()`'s one after another in one expression

Reply:
done, also extracted the expectations logic to separate type (to maybe use it in rs-sdk)